src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,340 @@
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						//return result;
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numDeaths[temp_date] = number;
+				if(allUSPlaces[i].numDeaths[temp_date] == GetMostRecentValue(allUSPlaces[i].numDeaths)){
+					allUSPlaces[i].currentNumDeaths=number;
+				}//check if the number is the most recent number
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numDeaths) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numConfirmed[temp_date] = number;
+				if(allUSPlaces[i].numConfirmed[temp_date] == GetMostRecentValue(allUSPlaces[i].numConfirmed)){
+					allUSPlaces[i].currentNumConfirmed=number;
+				}
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+				break; //just in case
+			}
+		}
+	}
+	
+
+	return errormsg;
+}
+
+function EditUSData(county, state, date, tod, number) {
+    if(allUSPlaces == undefined){
+        InitializeAllUSPlace();
+    }
+    var result = false;
+	if (tod == 1) {
+		console.log(allUSPlaces.length);
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numConfirmed) {
+					var temp = new Date(key);
+					var d =
+						temp.getMonth() +
+						1 +
+						'/' +
+						temp.getDate() +
+						'/' +
+						temp.getFullYear();
+					if (d == date) {
+						allUSPlaces[i].numConfirmed[key] = number;
+						result = true;
+						break;
+					}
+				}
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numDeaths) {
+					var temp = new Date(key);
+					var d =
+						temp.getMonth() +
+						1 +
+						'/' +
+						temp.getDate() +
+						'/' +
+						temp.getFullYear();
+					if (d == date) {
+						allUSPlaces[i].numDeaths[key] = number;
+						result = true;
+						break;
+					}
+				}
+			}
+		}
+	}
+	return result;
+}
+
+
+function RemoveUSData(county, state, date, tod) {
```

Comment:
We want to update currentNumConfirmed and currentNumDeaths inside this function, just like inside AddUSData() function.